lumberdev/sum-it-up
# feat: add in-depth analytics

## WHAT IT DOES:
_Description of the pull request, what changed_
Adds Analytics to site.
Adds a new hook `useAnalytics` and a new file `analytics/events` which holds constants of analytic event names.

New tracking analytics calls have been added for almost every interaction on the site.

## HOW TO TEST:
_Description of how to test if applicable_

Navigate to the page.
Check the network tab.
Check to see that analytics are being fired.

<img width="160" alt="Screen Shot 2023-02-15 at 6 50 03 PM" src="https://user-images.githubusercontent.com/90791391/219221333-51909d36-c3f5-41d0-bc40-62e0d8712971.png">

Alternatively you could install the chrome extension for easy debugging
https://chrome.google.com/webstore/detail/segment-event-tracker/hbanigoffkilibdakdmmlgefndpjmajl?hl=en

## WHAT'S LEFT
- [ ] - Consent manager if necessary
- [x] - GA hook up in Segment
- [ ] - Add share button analytics once implemented
